File: FrontendApp/src/app/models/dtos/task-info.dto.ts
import { TaskStatus } from "../../enums/task-status.enum";
import { UserDTO } from "./user.dto";

export interface TaskInfoDTO {
  id: number;
  companyId: number | null;
  companyName: string | null;
  parentId: number | null;
  name: string | null;
  description: string | null;
  budget: number | null;
  status: TaskStatus | null;
  completeDate: string | null;
  members: UserDTO[] | null;
}

File: Services/TaskService/TaskService.Application/DTOs/TaskInfoDTO.cs
namespace TaskService.Application.DTOs
{
    public class TaskInfoDTO
    {
        public int Id { get; set; }
        public int? CompanyId { get; set; }
        public string? CompanyName { get; set; }
        public int? ParentId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal? Budget { get; set; }
        public Domain.Enums.TaskStatus? Status { get; set; }
        public DateOnly? CompleteDate { get; set; }
        public UserDTO[]? Members { get; set; }
    }
}